### PR TITLE
Added possibility to change prompt by global flag

### LINKED ISSUE #477: Make iotile tool more friendly for programmatic control
Right now it's difficult to programmatically determine when the `iotile` tool has finished processing a command since the only clue is that it prompts with a context name for another input. 

There should be a global command line flag that allows you to turn off that context specific prompt and replace it with a user specified giant random string as in http multipart form uploads.  This string should then be echoed back after the context name to provide a definitive trigger for when iotile has finished work:

```
iotile -d "context_marker"
(root)context_marker
```

In particular:
- [ ] add a `-d` global flag to the iotile tool, that if passed will cause that string to be printed after the context name.
- [ ] When `-d` is specified, no space should be included between `(context)<DELIMITER>` or after delimiter to allow for easy, unambiguous machine parsing of the context name and the command output of the previous line.

### PR DESCRIPTION
**Pull request by vvynohradov**
Closes #477
## Overview

Added possibility to change prompt by global flag

## Major Changes

1. Added possibility to change prompt
2. Added tests for shell prompt

### CODE REVIEW

File: iotilecore/iotile/core/scripts/iotile_script.py [hunk 1]
```diff
@@ -231,7 +233,7 @@ def main(argv=None):
     try:
         while True:
             try:
-                linebuf = input("(%s) " % shell.context_name())
+                linebuf = input(re.sub(r"(?<!{){context}(?!})", shell.context_name(), args.prompt))
```
Comment (vvynohradov) [inline iotilecore/iotile/core/scripts/iotile_script.py]: Maybe we should change this to use .format

**Comment by mattrunchey** [comment]
- Typo in your test file name

Seems like your test case isn't passing, from the test matrix:
```---------------------------- Captured stderr call -----------------------------

usage: pytest [-h] [-v] [-l LOGFILE] [-i INCLUDE] [-e EXCLUDE] [-q]

              [-t TIMEOUT]

              ...

pytest: error: unrecognized arguments: -p```
from `

>       main(['-p', '{context} 123'])`

**Comment by vvynohradov** [comment]
> * Typo in your test file name (`test_promt` to `test_prompt`)
> 
> Seems like your test case isn't passing, from the test matrix:
> 
> ```
> ---------------------------- Captured stderr call -----------------------------
> 
> usage: pytest [-h] [-v] [-l LOGFILE] [-i INCLUDE] [-e EXCLUDE] [-q]
> 
>               [-t TIMEOUT]
> 
>               ...
> 
> pytest: error: unrecognized arguments: -p
> ```
> from
> `> main(['-p', '{context} 123'])`

Hello @mattrunchey , 
Thank you for your reply. Can you please try to run `iotile -p` to see if this flag exists and you have latest module version? Maybe, you received error when iotilecore was installing and iotilecore module didn't install.

**Comment by mattrunchey** [comment]
> > * Typo in your test file name (`test_promt` to `test_prompt`)
> > 
> > Seems like your test case isn't passing, from the test matrix:
> > ```
> > ---------------------------- Captured stderr call -----------------------------
> > 
> > usage: pytest [-h] [-v] [-l LOGFILE] [-i INCLUDE] [-e EXCLUDE] [-q]
> > 
> >               [-t TIMEOUT]
> > 
> >               ...
> > 
> > pytest: error: unrecognized arguments: -p
> > ```
> > from
> > `> main(['-p', '{context} 123'])`
> 
> Hello @mattrunchey ,
> Thank you for your reply. Can you please try to run `iotile -p` to see if this flag exists and you have latest module version? Maybe, you received error when iotilecore was installing and iotilecore module didn't install.

I didn't run the command. This is failing in the automatic tests https://dev.azure.com/iotile/coretools/_build/results?buildId=105

**Comment by mattrunchey** [comment]
Going to close this due to inactivity; we can reopen when there is time to action the changes needed.

**Pull request closed without merging**